### PR TITLE
chore: unify effect set syntax

### PR DESCRIPTION
**Pull request by mlutze**
This removes the `and` syntax from the parser. `+` now means `and`, and `&` means `or`. 

After this PR we should be able to re-enable most of the disabled tests,

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 1]
```diff
@@ -3088,6 +3177,7 @@ object Weeder {
     * Performs weeding on the given effect `ident`.
     * Checks whether it is actually the keyword `Static`.
     */
+  // TODO remove
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:3180]: TODO?

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 2]
```diff
@@ -341,17 +338,26 @@ object Weeder {
...
+          val tVal = visitType(t0)
+          mapN(tVal) {
+            case t => Map(ident -> WeededAst.Case(ident, t, mkSL(sp1, sp2)))
+          }
         // Case 3: multiton enum
         case (None, Some(cs0)) =>
           /*
            * Check for `DuplicateTag`.
            */
+          // TODO clean up
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:351]: TODO

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 3]
```diff
@@ -490,15 +512,17 @@ object Weeder {
...
       val doc = visitDoc(doc0)
       val modVal = visitModifiers(mod0, legalModifiers = Set(Ast.Modifier.Public))
-      val argVal = args0.map(visitType).toList match {
+      val argVal0 = args0.toList match {
         case hd :: Nil => hd.toSuccess
-        case ts => NonUnaryAssocType(ts.length, ident.loc).toFailure
+        case ts => (NonUnaryAssocType(ts.length, ident.loc): WeederError).toFailure
       }
-      val tpe = visitType(tpe0)
+      // TODO avoid short-circuiting with flatMap
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:519]: TODO

**Comment by magnus-madsen** [comment]
Ill merge this so we can progress :)

**Comment by magnus-madsen** [comment]
@stephentetley This changes the effect syntax, as discussed here: #5858

**Comment by stephentetley** [comment]
Thanks for the headsup Magnus!

**Pull request merged**